src/services_models/services_constructor.jl

```diff
@@ -39,6 +40,17 @@ function construct_services!(
             )
         end
     end
+
+    groupservice = nothing
+
+    for (key, service_model) in services_template
+        if service_model.formulation === GroupReserve  # group service needs to be constructed last
+            groupservice = key
+            continue
+        end
+        _construct_valid_services!(service_model)
+    end
+    groupservice === nothing || _construct_valid_services!(services_template[groupservice])
```

Comment:
Is this the standard Julian way of writing

```
if !isnothing(groupservice)
    _construct_valid_services!(services_template[groupservice])
end
```
?

Reply:
I believe so. @nickrobinson251 ?

Reply:
yeah, short circuit "and" `&&` and "or" `||` are used this way a lot in Julia. 
But it is exactly equivalent to ```julia
if groupservice !== nothing
    _construct_valid_services!(services_template[groupservice])
end
```

Similar to how `a ? x : y` is used instead of `if a; x; else y; end`

However `a === nothing` is sometimes recommended over `isnothing(a)` for [some reason to do with the compiler].

Reply:
I can get used to it if it's standard practice.